src/hardware/ad7172.rs

```diff
@@ -42,8 +42,50 @@ pub enum AdcReg {
     GAIN3 = 0x3b,
 }
 
-// ADC SETUPCON register settings.
-struct Setupcon;
+// *Note*: Register bitfields are not exhaustive.
+
+// ADC MODE register settings.
```

Comment:
Stick to the names in the datasheet.

```suggestion
// ADCMODE register settings.
```

And make these `///` docstrings if they are intended to be.
It doesn't make sense to have the empty struct `pub` but not its associated constants. Either both or none.
Finally it's much less confusing if you stick to the value size of the registers (`u16` here).